common.go

```diff
@@ -17,3 +18,20 @@ func debugLog(format string, args ...interface{}) {
 	}
 	log.Printf(format+"\n", args...)
 }
+
+//logURL replaces existing password from the given URL
+func logURL(URL string) (string, error) {
```

Comment:
1. Maybe call this "filteredURL"?
2. URL should be lower-case, but that collides with the package name. Just call this u; the function name carries the necessary context.